Lib/test/test_capi/test_misc.py

```diff
@@ -401,6 +401,43 @@ def check_negative_refcount(self, code):
                          br'_Py_NegativeRefcount: Assertion failed: '
                          br'object has negative ref count')
 
+    def run_iter_api_test(self, next_func):
+        dataset = (
+            (), (1,2,3),
+            [], [1,2,3],
```

Comment:
Can you test an "AIter"?

Maybe test also an iter object: `iter([1, 2, 3])`.

Reply:
That's what we do a few lines below.